### PR TITLE
[vividus] Align syntax of init variable step

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 1]
```diff
@@ -121,6 +121,31 @@ When I wait `$period` for debug
 When I wait `PT30S` for debug
 ----
 
+=== Initialize variable with value
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:124]: such step should be first in the docs

**Comment by ikalinin1** [comment]
either separate the commits or put the correct name

File: vividus/src/main/resources/steps/defaults/composite.steps [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Composite: When I initialize the $scopes variable `$variableName` with value `$variableValue`
+Priority: 1
+!-- WARNING: The step "When I initialize $scopes variable `$variableName` with value `$variableValue`" is deprecated and will be removed in VIVIDUS 0.6.0
```
Comment (ikalinin1) [inline vividus/src/main/resources/steps/defaults/composite.steps:3]:
```suggestion
!-- WARNING: The step "When I initialize the $scopes variable `$variableName` with value `$variableValue`" is deprecated and will be removed in VIVIDUS 0.6.0
```

File: vividus/src/test/java/org/vividus/steps/VariablesStepsTests.java [hunk 3]
```diff
@@ -106,10 +106,10 @@ void testCompareSimpleVariablesStrings(String variable1, String variable2, boole
     }
 
     @Test
-    void testInitVariableWithGivenValue()
+    void shouldInitiialzeVariableWithValue()
```
Comment (ikalinin1) [inline vividus/src/test/java/org/vividus/steps/VariablesStepsTests.java:109]: useless changes?

Comment (uarlouski) [reply vividus/src/test/java/org/vividus/steps/VariablesStepsTests.java:109]: intentional

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 4]
```diff
@@ -4,6 +4,31 @@
 
 Here one could find description of the steps that are delivered with Vividus itself without any plugins required.
 
+=== Initialize variable with value
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:7]:
```suggestion
=== Initialize variable
```

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 5]
```diff
@@ -18,11 +43,11 @@ When I execute steps:$stepsToExecute
 .Check and increment the value of a variable
 [source,gherkin]
 ----
-When I initialize the STORY variable `var` with value `0`
+When I initialize STORY variable `var` with value `0`
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:46]:
```suggestion
When I initialize story variable `var` with value `0`
```
let's use lower-case notation across all the docs

Comment (valfirst) [reply docs/modules/commons/pages/vividus-steps.adoc:46]: and the same for tests

File: vividus-tests/src/main/resources/story/integration/JSON - Wait for data in HTTP response.story [hunk 6]
```diff
@@ -17,6 +17,18 @@ Scenario: Verify step 'When I save JSON element from `$json` by JSON path `$json
 When I save JSON element from `${response}` by JSON path `<jsonPath>` to scenario variable `numberOfBooks`
 Then `${numberOfBooks}` is equal to `<booksNumber>`
 
+Scenario: Verify step "Then JSON element from `$json` by JSON path `$jsonPath` is equal to `$expectedData`$options"
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/JSON - Wait for data in HTTP response.story:20]: I guess this scenario was moved to `JSON.story` and this change was introduced due to invalid merge conflict resolution

File: vividus-tests/src/main/resources/story/integration/JSON - Wait for data in HTTP response.story [hunk 7]
```diff
@@ -17,6 +17,18 @@ Scenario: Verify step 'When I save JSON element from `$json` by JSON path `$json
...
+When I initialize scenario variable `current-date` with value `#{generateDate(P, yyyy-MM-DD)}`
+Given I initialize the scenario variable `expected-json` using template `/data/json-validation-template.ftl` with parameters:
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/JSON - Wait for data in HTTP response.story:22]: I think this step should be updated as well

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 8]
```diff
@@ -4,6 +4,31 @@
 
 Here one could find description of the steps that are delivered with Vividus itself without any plugins required.
 
+=== Initialize variable with value
+
+Initializes a variable with the specified value.
+
+[source,gherkin]
+----
+When I initialize $scopes variable `$variableName` with value `$variableValue`
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:13]: `When` or `Given` ?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/commons/pages/vividus-steps.adoc (hunk 8 above)
Comment (uarlouski) [reply docs/modules/commons/pages/vividus-steps.adoc:13]: well, Given seems to be more logically correct

File: vividus/src/main/java/org/vividus/steps/VariablesSteps.java [hunk 9]
```diff
@@ -92,8 +93,8 @@ protected boolean compareListsOfMaps(Object variable1, Object variable2)
      * @param variableName A name under which the value should be saved
      * @param variableValue A value to be saved
      */
-    @When("I initialize the $scopes variable `$variableName` with value `$variableValue`")
-    public void initVariableWithGivenValue(Set<VariableScope> scopes, String variableName, String variableValue)
+    @Given("I initialize $scopes variable `$variableName` with value `$variableValue`")
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/steps/VariablesSteps.java:96]: can we have an alias with "when"? :)

**Review by ikalinin1 — APPROVED** [review]

File: vividus/src/main/java/org/vividus/steps/VariablesSteps.java (hunk 9 above)
Comment (uarlouski) [reply vividus/src/main/java/org/vividus/steps/VariablesSteps.java:96]: @valfirst can we?

**Pull request merged**